src/Main/Java/product/Product.java

```diff
@@ -41,11 +41,17 @@ public void setStorage(double storage) {
         this.storage = storage;
     }
 
-    public void setPurchasePrice(double purchasePrice) {
+    public void setPurchasePrice(double purchasePrice) throws Exception {
+        if (purchasePrice <= 0) {
+            throw new Exception("Valore negativo");
```

Comment:
Per avere un messaggio del genere dovremmo considerare solo i numeri minori di 0 , visto che prendiamo in esame i casi in cui è uguale a 0 sarebbe megio adattare il messaggio